snake_game.py

```diff
@@ -78,73 +75,150 @@
-            else:
-                pygame.draw.rect(canva, COLOUR_SNAKE, block_rect)
-            
     @property
     def head(self):
         return self.body[-1]
-    
+
     @property
     def length(self):
         return len(self.body)
-    
+
+    def draw_score(self):
+        score_surface = pygame.font.Font(
+            os.path.join(package_base_path, "Assets", "Font", "PoetsenOne-Regular.ttf"), 25
+        ).render(f"Count: {str(self.length - 3)}", True, (56, 74, 12))
+        score_rect = score_surface.get_rect(
+            center=(
+                int(CELL_NUMBER * CELL_SIZE - 60),
+                int(CELL_NUMBER * CELL_SIZE - 40),
+            )
+        )
+        canva.blit(score_surface, score_rect)
+
+    def play_sound(self):
+        self.crunch_sound_graphic.play()
+
+    def draw(self):
+        self.update_head_graphic()
+        self.update_tail_graphic()
+        for index, block in enumerate(self.body):
+            block_rect = pygame.Rect(
+                block.x * CELL_SIZE, block.y * CELL_SIZE, CELL_SIZE, CELL_SIZE
+            )
+            if index == 0:
+                canva.blit(self.tail_graphic, block_rect)
+            elif index == self.length - 1:
+                canva.blit(self.head_graphic, block_rect)
+            else:
+                next_block = self.body[index - 1] - block
+                prev_block = self.body[index + 1] - block
+                if prev_block.x == next_block.x:
+                    canva.blit(self.body_vertical_graphic, block_rect)
+                elif prev_block.y == next_block.y:
+                    canva.blit(self.body_horizontal_graphic, block_rect)
+                else:
+                    if (
+                        prev_block.x == -1
+                        and next_block.y == -1
+                        or prev_block.y == -1
+                        and next_block.x == -1
+                    ):
+                        canva.blit(self.body_tl_graphic, block_rect)
+                    elif (
+                        prev_block.x == -1
+                        and next_block.y == 1
+                        or prev_block.y == 1
+                        and next_block.x == -1
+                    ):
+                        canva.blit(self.body_bl_graphic, block_rect)
+                    elif (
+                        prev_block.x == 1
+                        and next_block.y == -1
+                        or prev_block.y == -1
+                        and next_block.x == 1
+                    ):
+                        canva.blit(self.body_tr_graphic, block_rect)
+                    elif (
+                        prev_block.x == 1
+                        and next_block.y == 1
+                        or prev_block.y == 1
+                        and next_block.x == 1
+                    ):
+                        canva.blit(self.body_br_graphic, block_rect)
+
+    def update_tail_graphic(self):
```

Comment:
Can you use a variable to represent self.body[1] - self.body[0]? And also fix the head direction issue.